lexxpluss_apps/src/board_controller.cpp

```diff
@@ -1179,6 +1184,46 @@ class state_controller { // Variables Implemented
                 }
             }
             break;
+        case POWER_STATE::EMERGENCY: {
+            wheel_relay_control();
+            auto psw_state{psw.get_state()};
+            if (!dcdc.is_ok() || psw_state == power_switch::STATE::LONG_PUSHED) {
+                set_new_state(POWER_STATE::OFF);
+            } else if (mbd.is_dead()) {
+                set_new_state(wait_shutdown ? POWER_STATE::TIMEROFF : POWER_STATE::LOCKDOWN);
+            } else if (psw_state == power_switch::STATE::PUSHED || mbd.power_off_from_ros() || !bmu.is_ok()) {
+                if (wait_shutdown) {
+                    if ((k_uptime_get() - timer_shutdown)> 60000) {
+                        set_new_state(POWER_STATE::OFF);
+                        dcdc.set_enable(false);
+                        psw.reset_state();
+                        esw.reset_state();
+                    }
+                } else {
+                    LOG_DBG("wait shutdown\n");
+                    wait_shutdown = true;
+                    timer_shutdown = k_uptime_get();    // timer reset
+                    if (psw_state == power_switch::STATE::PUSHED)
+                        shutdown_reason = SHUTDOWN_REASON::SWITCH;
+                    if (mbd.power_off_from_ros())
+                        shutdown_reason = SHUTDOWN_REASON::ROS;
+                    if (!bmu.is_ok())
+                        shutdown_reason = SHUTDOWN_REASON::BMU;
+                    // Set LED
+                    led_controller::msg msg_led;
+                    msg_led.pattern = led_controller::msg::SHOWTIME;
+                    msg_led.interrupt_ms = 0;
+                    while (k_msgq_put(&led_controller::msgq, &msg_led, K_NO_WAIT) != 0)
+                        k_msgq_purge(&led_controller::msgq);
+                }
+            } else if (!esw.is_asserted() && !mbd.emergency_stop_from_ros() && mbd.is_ready()) {
+                LOG_DBG("not emergency and heartbeat OK\n");
+                set_new_state(POWER_STATE::RESUME_WAIT);
+            }
+            break;
+        }
+        case POWER_STATE::RESUME_WAIT:
+            set_new_state(POWER_STATE::NORMAL);
```

Comment:
Add some rules in [this feature](https://lexxpluss.atlassian.net/browse/HWDEV-2001)